pip/compat.py

```diff
@@ -41,7 +46,14 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            if WINDOWS:
+                try:
+                    from ctypes import cdll
+                    return s.decode("cp" + str(cdll.kernel32.GetACP()))
+                except ImportError:
```

Comment:
Hmm, I hadn't thought about IronPython (or jython for that matter). Quite possibly ctypes isn't available there.